File: src/client/packages/stashed_api/lib/src/models/media.dart
import 'dart:typed_data';

import 'package:equatable/equatable.dart';
import 'package:stashed_api/src/converters/uint8_list_json_converter.dart';

abstract class Media extends Equatable {
  final String sha256;
  final String md5;
  final String mimeType;
  final String? filename;
  final int size;
  @Uint8ListJsonConverter()
  final Uint8List? content;

  const Media({
    required this.sha256,
    required this.md5,
    required this.mimeType,
    this.filename,
    required this.size,
    this.content,
  });

  @override
  List<Object?> get props => [sha256, mimeType, filename, content == null];
}

File: src/server/Common/Entities/Media.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Stashed.Common.Entities;

public abstract class Media
{
    [Key] public required string Sha256 { get; init; }
    [Required] public required string Md5 { get; init; }
    [Required] public required string MimeType { get; init; }
    public required string? FileName { get; init; }
    [Required] public required int Size { get; init; }
    [Required] public required int Width { get; init; }
    [Required] public required int Height { get; init; }
    public required List<Tag> Tags { get; init; }
    [NotMapped] public required byte[] Content { get; set; }
}